source/incqueryserver-jupyter/iqs_jupyter/oicd_client.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 IncQuery Labs Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from json import JSONDecodeError
+
+import urllib3
+
+from iqs_client.api_client import ApiClient
+from iqs_client.configuration import Configuration
+
+
+class OicdClient(ApiClient):
```

Comment:
For me it seems like almost the same code as in `ApiClientWithOIDC`. If this modification contains fixes/enhancements for the original implementation we shall improve the original one and use it in this package, since it is not a jupyter-related addition.

Reply:
it's a slightly modified version of ApiClientWithOIDC. It differs in two places; the exception handling around the response, which the original didn't have but needed; and the constructor parameters.
To be used with the iqs_jupyter library, the api needs to be pieced together in decorate_client, which I also modified to work with different client classes.

Reply:
Moved the api client back to the iqs repo; see IncQueryLabs/incquery-server#4839